### ISSUE TITLE
Project save not triggered until the changes you're drawing are not active

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1268]
Potentially similar to the issue with sound clips we saw a while ago (#1255)?

**TL;DR** If you edit a backdrop, then leave the page, you won't necessarily be prompted to save your changes, and the project won't necessarily be auto-saved.

Chrome 55, Mac OS El Capitan, Production

**To repro:**
- See Inside a project so you're in the editor
- Click a backdrop
- Click backdrops tab
- Choose a shape from the menu
- Add the shape to the backdrop
- Go to My Stuff
- Note you are not asked to save changes
- See Inside the project
- Click the backdrop
- Go to backdrops tab
- Note that the backdrop has not been updated

If you do the same thing again, but click somewhere else in the editor e.g. the tool bar for the paint editor after you add the new shape to the backdrop, then you will **either prompt a save immediately** (in the editor), or you will mark the project as needs-save so **you'll be asked about your changes when you navigate away** from the page.

(It's hard to see what exactly is happening in this gif b/c the end and beginning look the same - but the blue ellipse is not being saved when I navigate away from the project back to My Stuff.)
![out](https://cloud.githubusercontent.com/assets/1320456/22210801/fd39390e-e158-11e6-9345-e57772b7bda1.gif)
